### PR TITLE
(#109) Log errors correctly

### LINKED ISSUE #109: Errors being logged incorrectly
### Issue description
A number of locations in `ESAutosuggestQueryService` and `ESTermsQueryService` call the logger with code similar to

```csharp
            catch (Exception ex)
            {
                String msg = build the message;
                _logger.LogError(msg, ex);
                throw new APIErrorException(500, msg);
            }
```

This form of the `LogError` call is for passing parameters into a formatting string, not for passing an exception.  The correct form is

```csharp
LogError (this Microsoft.Extensions.Logging.ILogger logger, Exception exception, string message, params object[] args);
```

> **ESTIMATE** 20

### Steps to reproduce the issue

1.  Run a test which results in an exeption

### What's the expected result?
- Details of the exception should appear in the log.

### What's the actual result?
- No details.

### PR DESCRIPTION
**Pull request by blairl-nih**
- Specify exception in the correct location in the LogError call when logging an exception.
- Replace String.Format calls with inline interpolation.
- Switch to LogDebug when logging normal no result.

Close #109

### CODE REVIEW

File: src/NCI.OCPL.Api.Glossary/Services/ESAutosuggestQueryService.cs [hunk 1]
```diff
@@ -81,15 +81,14 @@ public async Task<Suggestion[]> GetSuggestions(string dictionary, AudienceType a
             }
             catch (Exception ex)
             {
-                string msg = String.Format("Could not search dictionary '{0}', audience '{1}', and language '{2}'.", dictionary, audience, language);
-                _logger.LogError(msg, ex);
+                string msg = $"Could not search dictionary '{dictionary}', audience '{audience}', and language '{language}'.";
```
Comment (bryanpizzillo) [inline src/NCI.OCPL.Api.Glossary/Services/ESAutosuggestQueryService.cs:84]: Please log the index name that is failing as well

File: src/NCI.OCPL.Api.Glossary/Services/ESTermsQueryService.cs [hunk 2]
```diff
@@ -66,21 +66,21 @@ public async Task<GlossaryTerm> GetById(string dictionary, AudienceType audience
             }
             catch (Exception ex)
             {
-                String msg = String.Format("Could not search dictionary '{0}', audience '{1}', language '{2}' and id '{3}.", dictionary, audience, language, id);
-                _logger.LogError(msg, ex);
+                String msg = $"Could not search dictionary '{dictionary}', audience '{audience}', language '{language}' and id '{id}.";
```
Comment (bryanpizzillo) [inline src/NCI.OCPL.Api.Glossary/Services/ESTermsQueryService.cs:69]: Please log the index name as well

File: src/NCI.OCPL.Api.Glossary/Services/ESTermsQueryService.cs [hunk 3]
```diff
@@ -200,16 +200,15 @@ public async Task<GlossaryTermResults> GetAll(string dictionary, AudienceType au
             }
             catch (Exception ex)
             {
-                String msg = String.Format("Could not get dictionary '{0}', audience '{1}', language '{2}', size '{3}', from '{4}'.", dictionary, audience, language, size, from);
+                String msg = $"Could not get dictionary '{dictionary}', audience '{audience}', language '{language}', size '{size}', from '{from}'.";
                 _logger.LogError("Error Fetching All from index: " + this._apiOptions.AliasName);
```
Comment (bryanpizzillo) [inline src/NCI.OCPL.Api.Glossary/Services/ESTermsQueryService.cs:204]: Can you change this to the formatter please

Comment (bryanpizzillo) [reply src/NCI.OCPL.Api.Glossary/Services/ESTermsQueryService.cs:204]: This should be for the line, `_logger.LogError("Error Fetching All from index: " + this._apiOptions.AliasName);`. GH review tool is adding this someplace odd...

**Comment by blairlearn** [comment]
@bryanpizzillo - changes are in.

**Comment by blairlearn** [comment]
@bryanpizzillo's changes went in with the force push. :-)

**Comment by blairlearn** [comment]
Found a few more places that were still using long-winded String.Format instead of the more concise `$"{variable}"` syntax.

**Pull request merged**